core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/util/Triple.java

```diff
@@ -0,0 +1,15 @@
+package org.optaplanner.core.impl.util;
+
+public interface Triple<A, B, C> {
```

Comment:
Can we add a comment on Triple and Pair on how they implement equals/hashcode?

We should also probably specify for UniTuple, BiTuple and TriTuple that they don't overwrite equals/hashcode (deliberately) if they aren't already.